### ISSUE TITLE
Public link share expiry interaction with core

### ISSUE THREAD

**Issue by phil-davis** [owncloud/password_policy#286]
The password policy public link expiration settings can conflict with core settings, e.g. link expiry set to 5 days in password policy:
![Screenshot_2020-03-11 Settings - ownCloud(1)](https://user-images.githubusercontent.com/1535615/76381151-c62aa700-637c-11ea-97b4-9d5326d24afa.png)
Link expiry set to 7 days in core:
![Screenshot_2020-03-11 Settings - ownCloud](https://user-images.githubusercontent.com/1535615/76381197-e0648500-637c-11ea-981c-c0aa59d81bb2.png)
Create a public link:
![Screenshot_2020-03-11 Files - ownCloud](https://user-images.githubusercontent.com/1535615/76381221-ee1a0a80-637c-11ea-9bce-25860966b303.png)

core fills in the default expiry date, then when you click "Share" to create the link, password policy in the backend complains that "The expiration date cannot exceed 5 days."

**Comment by phil-davis** [comment]
This problem is "just a nuisance" - the user can adjust the expiration date earlier and the link gets created.

**Comment by phil-davis** [comment]
Someone remind me - what is the real point of the settings in password policy vs core?

In password policy you can set different maximum expiration depending if a password is set on the public link share or not. Is that the only extra functionality that is available?

**Comment by pako81** [comment]
https://github.com/owncloud/password_policy/issues/286#issue-578994242 caused by a missing check for the share type (public link) here https://github.com/owncloud/password_policy/blob/master/lib/HooksHandler.php#L201.

So the proposed fix would need to add two additional checks: one at the above-mentioned place, the other one at https://github.com/owncloud/password_policy/blob/master/lib/HooksHandler.php#L195, which will solve https://github.com/owncloud/password_policy/issues/287.

So one fix for two issues: 1. fix preventing ordinary shares when public link expiry is set and 2. fix conflicts between core and password_policy expiration policies.

**Comment by pako81** [comment]
> In password policy you can set different maximum expiration depending if a password is set on the public link share or not. Is that the only extra functionality that is available?

Yes, it is. IMHO we should just get rid of the core settings since this us causing redundancy.

**Comment by phil-davis** [comment]
> Yes, it is. IMHO we should just get rid of the core settings since this is introducing redundancy.

The core settings let you specify a default expiration date without enforcing it (and you can enforce it if you want)

Maybe define a combined set of requirements for the feature, then decide if the combined requirements are implemented in core (and remove existing link expiration settings from password policy) or the reverse.

**Comment by phil-davis** [comment]
I tested the above with core 10.3.2 - the "annoying" behaviour is the same.
So this is not a regression in core 10.4.0.

IMO the "annoyance" can be improved in some future software version (of core and/or password_policy, wherever the code needs to be enhanced). That work can scheduled as normal.

**Comment by VicDeo** [comment]
> In password policy you can set different maximum expiration depending if a password is set on the public link share or not. Is that the only extra functionality that is available?
Yes, it is. IMHO we should just get rid of the core settings since this is introducing redundancy.

I think reusing of the core setting in the app is the better approach.
